python_codes/contour_plots.py

```diff
@@ -0,0 +1,135 @@
+#! /usr/bin/env python3
+from pylab import *
+from netCDF4 import Dataset
+import numpy as np
+
+'''Retrieving Data from the nc files'''
+data = Dataset('/home/mafields/CylconeDiagnostics/data/jupiter2d-long-main.nc')
+
+'''Defining Variables'''
+time = data['time']
+pressure =data['press']
+rho_gas = data['rho'][:,:,:,0]
+temp = data['temp']
+print(temp.shape)
+#mass = rho_gas*volume
+
+horiz_v1 = data['vel1'][:,:,:,0]
+horiz_v2 = data['vel2'][:,:,:,0]
+
+print(horiz_v1.shape)
+
+kinetic_energy = (1/2)*(horiz_v1**2+horiz_v2**2)
+print(kinetic_energy)
+print(kinetic_energy.shape)
+
+#print(time.shape)
+#print(pressure.shape)
+
+H2Ov = data['vapor1'][:,:,:,0]
+H2Oc = data['H2Op1'][:,:,:,0]
+H2Op = data['H2Op2'][:,:,:,0]
+
+#print(H2Ov.shape)
+#print(H2Oc.shape)
+#print(H2Op.shape)
+
+NH3v = data['vapor2'][:,:,:,0]
+NH3c = data['NH3p1'][:,:,:,0]
+NH3p = data['NH3p2'][:,:,:,0]
+
+dry_air = rho_gas*H2Ov + rho_gas*NH3v
+H2O_total = rho_gas*H2Ov + H2Oc + H2Op
+NH3_total = rho_gas*NH3v + NH3c + NH3p
+
+'''Finding total mass'''
+dry_air_t = sum(dry_air, axis = (1,2))
+H2Ot = sum(H2O_total, axis = (1,2))
+NH3t = sum(NH3_total, axis = (1,2))
+
+
+'''Deviations of Temperature and Concentrations'''
+
+'''temperature'''
+
+t_0 = np.zeros((len(temp[0,0,:,0])))
+
+for i in range(len(temp[0,0,:,0])): 
+    t_0 = temp[0,0,i,0]
+print(t_0)
+
+temp_variations = np.zeros((len(temp[:,0,0,0]),len(temp[0,:,0,0]),len(temp[0,0,:,0])))
+for i in range(len(temp[:,0,0,0])):
+    for j in range(len(temp[0,:,0,0])):
+        for k in range(len(temp[0,0,:,0])):
+            temp_variations[i,j,k] = temp[i,j,k,0] - t_0
```

Comment:
These can be vectorized as well:
```
ntime, nx1, nx2, nx3 = temp.shape
temp_variations = temp - t_0.reshape((1,nx1,1,1))
```

Reply:
what does this mean

Reply:
It is called [broadcasting](https://www.geeksforgeeks.org/python-broadcasting-with-numpy-arrays/).